core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java

```diff
@@ -1056,6 +1089,9 @@ public String getPackageModules(String packageName,
                                     String packageVersion,
                                     int offset,
                                     int limit) {
+
+        if(!assertPackageExistence(packageName, packageVersion)) return null;
```

Comment:
This way we need to run 2 queries whenever the package or modules are retrieved. This might affect the performance (not in a good way). I think a better solution be to prepend the `count(<retrieved field>)` to select and just check if it is > 0 before returning the result and if <1 return `null`, i.e. in this case the query would be something like `select count(f.*), f.* from ....`.
Same applies to the other changed methods.

Reply:
> a better solution be to prepend the `count(<retrieved field>)`
I surely thought about this way, but it won't work as expected; an empty result can in fact happen.

> to run 2 queries whenever the package
But I completely agree that this sucks. That's why I asked you for advice on whether or not there is an alternative.

Reply:
> > a better solution be to prepend the `count(<retrieved field>)`
> 
> I surely thought about this way, but it won't work as expected; an empty result can in fact happen.
> 
> > to run 2 queries whenever the package
> 
> But I completely agree that this sucks. That's why I asked you for advice on whether or not there is an alternative.
> 
> Any thoughts?

Hm, yes, I guess you are right. Then will have to go with 2 queries for now as you implemented. Feel free to merge this when you increase the timeout. Thanks!

Reply:
> Feel free to merge this when you increase the timeout.

Moved to #277